.github/workflows/interop_tests.yml

```diff
@@ -62,30 +36,92 @@ jobs:
+        run: |
+          cd mls-rs/test_harness_integration
+          cargo run -- --port 50001 &
+          cargo build --no-default-features
+          cargo run --no-default-features -- --port 50002 &
+          cd ../../interop/interop
           test-runner/test-runner --client localhost:50001 --client localhost:50002 --suite 1 --public --config mls-rs-configs/bare_bones.json > /dev/null
-
-          # all features with "no tree_index"
-          for config in `ls mls-rs-configs | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50003 --suite 1 --config mls-rs-configs/$config ; done > /dev/null
-
-          # all features with "no private_message"
-          for config in `ls mls-rs-configs | grep -v "application"  | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50004 --public --suite 1 --config mls-rs-configs/$config ; done > /dev/null
-
-          # all features with "no prior_epoch"
-          for config in `ls mls-rs-configs | grep -Ev "(application_out_of_order_across)|(psk)" | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50005 --suite 1 --config mls-rs-configs/$config ; done > /dev/null
-
-          # all features with "no out_of_order"
-          for config in `ls mls-rs-configs | grep -v "application_out_of_order" | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50006 --suite 1 --config mls-rs-configs/$config ; done > /dev/null
-
-          # all features with "no psk"
-          for config in `ls mls-rs-configs | grep -Ev "(psk)|(branch)|(reinit)" | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50007 --suite 1 --config mls-rs-configs/$config ; done > /dev/null
-
-           # all features with "no custom_proposal"
-          for config in `ls mls-rs-configs | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50008 --suite 1 --config mls-rs-configs/$config ; done > /dev/null
-
-           # all features with "no by_ref_proposal"
-          for config in `ls mls-rs-configs | grep -E "(application)|(commit_by_value)|(branch)|(welcome_join)" | sed -e "s/mls-rs-configs\///"`; do >&2 echo $config && test-runner/test-runner --client localhost:50001 --client localhost:50009 --suite 1 --config mls-rs-configs/$config ; done > /dev/null
+          kill %1
+          kill %2
+      - name: Test interop; full feature with "no tree_index"
+        run: |
+          cd mls-rs/test_harness_integration
+          cargo run -- --port 50001 &
+          cargo build --no-default-features --features private_message,prior_epoch,out_of_order,psk,custom_proposal,by_ref_proposal
```

Comment:
Building before running should not be necessary, as `cargo run` triggers a build if the artifacts are out of date.

Reply:
It sure does, but `cargo run &` in the background doesn't return the exit code until you `wait` for the process. So, it does `cargo build` to get the exit code and kill the action on fail. Does this make sense?

Reply:
So is the build done first to ensure the CI job fails immediately if a build error occurs?

Reply:
Yes, exactly

Reply:
Makes sense, thank you. And given that it seems to start servers and clients, I guess you may run into connection issues if the server were to take too long to start because it needs to be built first.